modules/wayfire/hm.nix

```
@@ -1,66 +1,86 @@
 {
+  mkTarget,
   config,
   lib,
   pkgs,
   ...
 }:
-let
-  cfg = config.stylix.targets.wayfire;
-in
+mkTarget {
+  name = "wayfire";
+  humanName = "wayfire";
 
-{
-  options.stylix.targets.wayfire = {
-    enable = config.lib.stylix.mkEnableTarget "wayfire" true;
-    useWallpaper = config.lib.stylix.mkEnableWallpaper "wayfire" true;
-  };
-
-  config = lib.mkIf (config.stylix.enable && cfg.enable) (
-    let
-      inherit (config.lib.stylix) colors;
-      rgba = rgb: a: "\\#${rgb}${a}";
-      rgb = (lib.flip rgba) "ff";
+  extraOptions.useWallpaper = config.lib.stylix.mkEnableWallpaper "wayfire" true;
 
-      wayfireConfig = config.wayland.windowManager.wayfire;
+  configElements = [
+    (
+      { fonts }:
+      {
+        wayland.windowManager.wayfire.settings.decoration.font =
+          "${fonts.monospace.name} ${toString fonts.sizes.desktop}";
+      }
+    )
+    (
+      {
+        cfg,
+        image,
+        imageScalingMode,
+      }:
+      let
+        wayfireBackground = pkgs.runCommand "wayfire-background.png" { } ''
+          ${lib.getExe' pkgs.imagemagick "convert"} ${image} $out
+        '';
+      in
+      {
+        wayland.windowManager.wayfire.wf-shell.settings = {
+          background.image = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
+          cube = {
+            cubemap_image = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
+            skydome_texture = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
+          };
+          background.fill_mode =
+            if imageScalingMode == "stretch" then
+              "stretch"
+            else if imageScalingMode == "fit" then
+              "preserve_aspect"
+            else
+              "fill_and_crop";
 
-      wayfireBackground = pkgs.runCommand "wayfire-background.png" { } ''
-        ${lib.getExe' pkgs.imagemagick "convert"} ${config.stylix.image} $out
-      '';
-    in
-    {
-      wayland.windowManager.wayfire.settings = lib.mkIf wayfireConfig.enable {
-        cube = {
-          background = rgb colors.base00;
-          cubemap_image = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
-          skydome_texture = lib.mkIf cfg.useWallpaper (toString wayfireBackground);
         };
+      }
+    )
+    (lib.mkIf config.stylix.targets.nixos-icons {
```

Considering

https://github.com/nix-community/stylix/blob/3ca2c4478a1e984d2007c57467c6986bcdcb2629/modules/nixos-icons/overlay.nix#L8-L9

this should probably be:

```suggestion
    (lib.mkIf config.stylix.targets.nixos-icons.enable {
```